src/vos/evtree.c

```diff
@@ -2522,11 +2522,145 @@ evt_ssof_adjust(struct evt_context *tcx, struct evt_node *nd,
 	return offset;
 }
 
+/**
+ * Sorted by Start Offset (SSOF)
+ *
+ * Extents are sorted by start offset first, then high to low epoch, then end
+ * offset
+ */
+
+/** Rectangle comparison for sorting */
+static int
+evt_ssof_cmp_rect(struct evt_context *tcx, const struct evt_rect *mbr,
+		  const struct evt_rect *rt1, const struct evt_rect *rt2)
+{
+	return evt_rect_cmp(rt1, rt2);
+}
+
+static int
+evt_ssof_insert(struct evt_context *tcx, struct evt_node *nd,
+		umem_off_t in_off, const struct evt_entry_in *ent,
+		bool *changed)
+{
+	return evt_common_insert(tcx, nd, in_off, ent, changed,
+				 evt_ssof_cmp_rect);
+}
+
+static int
+evt_ssof_adjust(struct evt_context *tcx, struct evt_node *nd,
+		struct evt_node_entry *ne, int at)
+{
+	return evt_common_adjust(tcx, nd, ne, at, evt_ssof_cmp_rect);
+}
+
 static struct evt_policy_ops evt_ssof_pol_ops = {
 	.po_insert		= evt_ssof_insert,
 	.po_adjust		= evt_ssof_adjust,
-	.po_split		= evt_ssof_split,
-	.po_rect_weight		= evt_ssof_rect_weight,
+	.po_split		= evt_even_split,
+	.po_rect_weight		= evt_common_rect_weight,
+};
+
+/**
+ * Sorted by distances to sides of bounding box
+ */
+
+/** Rectangle comparison for sorting */
+static int64_t
+evt_mbr_dist(const struct evt_rect *mbr, const struct evt_rect *rect)
+{
+	int64_t ldist = rect->rc_ex.ex_lo - mbr->rc_ex.ex_lo;
+	int64_t rdist = mbr->rc_ex.ex_hi - rect->rc_ex.ex_hi;
```

Comment:
I'm not sure if this matches with what we discussed, but what in my mind is, 
``` 
// mbr: MBR of the full node, which is going to be split into node_left and node_right
// rect: one of the rectangles within the full node, we run the process for all rectangles within the node
// pseudo code will be like

evt_select_node(node_full, rect)
{
  mbr = node_full.get_mbr()
  dist_A = mbr.hi - rect.lo;
  dist_B = rect.hi - mbr.lo;
  if (dist_A > dist_B)
      return select_left_side;
  else
      return selelct_right_side;
}
```

Reply:
yep, that's what this does.   It calculates distance from right side and distance from left side and takes the difference between the two.

Reply:
Oh, wait, I see the difference.  In my code, I'm using the near boundary to decide.  In yours, you're using the far boundary.   but it will give same decision every time.

If a rectangle is centered to the right of the MBR, it's right and left edges will both be closer to the right boundary.

Proof: We can write MBR_width as left + rect_width + right

If left > right
    left + rect_width > right + rect_width
else
    left + rect_width <= right + rect_width

So in my caculation, I simply use the difference between left and right

if (left - right) < 0
    insert_left
else
    insert_right

Entries are sorted in their node by this difference (though the difference isn't stored so MBR changes don't affect the order of entries already in the node).

Reply:
indeed, they are essentially same